### PR TITLE
Add OOM howto

### PR DESCRIPTION
**Pull request by carmocca**
Preview: https://github.com/Lightning-AI/lit-gpt/blob/carmocca/oom-howto/howto/oom.md

### CODE REVIEW

File: howto/oom.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+it means that your GPU memory size wasn't big enough for the model and script configuration.
+
+Here's a few things you can try:
+
+### Reduce the micro batch size
+
+Adjust the `micro_batch_size = ...` variable in the fine-tuning and pretraining scripts. This variable determines the number of samples loaded per iteration.
+
+A smaller value will simply load less samples simultaneously. The minimum value is 1.
```
Comment (lantiga) [inline howto/oom.md:19]:
```suggestion
A smaller value will simply load fewer samples simultaneously. The minimum value is 1.
```

File: howto/oom.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+Experiment with different micro batch sizes to find a balance between memory consumption and computational efficiency. Smaller micro batch sizes consume less memory but may result in slower training convergence. Conversely, larger micro batch sizes require more memory but can accelerate training speed.
+
+### Try a different optimizer
+
+Our scripts use the [`AdamW` optimizer](https://pytorch.org/docs/main/generated/torch.optim.AdamW.html).
+It maintains 2 states for each trainable parameter of the model, meaning that the optimizer memory is double compared to
+an optimizer like [`SGD`](https://pytorch.org/docs/main/generated/torch.optim.SGD.html).
+You can try replacing it with your optimizer of choice that is lighter in memory requirements. keep in mind that different optimizers have distinct optimization behaviors, so it's essential to assess their impact on the training process and model performance.
```
Comment (lantiga) [inline howto/oom.md:28]: Can we provide suggestions for "cheaper" optimizers that work with LLMs?

Comment (lantiga) [reply howto/oom.md:28]:
```suggestion
You can try replacing it with your optimizer of choice that is lighter in memory requirements. Keep in mind that different optimizers have distinct optimization behaviors, so it's essential to assess their impact on the training process and model performance.
```

File: howto/oom.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+This suggestion is particularly relevant for pretraining, as the trainable parameters in the model represent a small
+subset of the total in the fine-tuning scripts.
+
+### Use lower precision
+
+Our scripts expose the `--precision` argument, this directly impacts the memory usage.
+
+Using true lower precision (`16-true`, `bf16-true`) reduces the memory usage by half compared to `32-true`, however,
+the model might start producing NaNs due to the limited range of representable values.
```
Comment (lantiga) [inline howto/oom.md:38]: Especially with `16-true`, or we're seeing a lot of this with `b16-true` as well?

**Review by lantiga — APPROVED** [review]
Looks great. You haven't mentioned activation checkpointing or offloading in the last section. Is it worth doing?

File: howto/oom.md (hunk 3 above)
Comment (carmocca) [reply howto/oom.md:38]: Both, at least for adapter

**Pull request merged**